src/realizations/catchment/Bmi_Fortran_Formulation.cpp

```diff
@@ -0,0 +1,183 @@
+}
+
+string Bmi_Fortran_Formulation::get_output_line_for_timestep(int timestep, std::string delimiter) {
+    // TODO: something must be added to store values if more than the current time step is wanted
+    // TODO: if such a thing is added, it should probably be configurable to turn it off
+
+    // TODO: for now, just get current value, and ignore the timestep param
+
+    std::string output_str;
+
+    for (const std::string& name : get_output_variable_names()) {
+        output_str += (output_str.empty() ? "" : ",") + std::to_string(get_var_value_as_double(name));
+    }
+    return output_str;
+}
+
+/**
+ * Get the model response for a time step.
+ *
+ * Get the model response for the provided time step, executing the backing model formulation one or more times as
+ * needed.
+ *
+ * Function assumes the backing model has been fully initialized an that any additional input values have been applied.
+ *
+ * The function throws an error if the index of a previously processed time step is supplied, except if it is the last
+ * processed time step.  In that case, the appropriate value is returned as described below, but without executing any
+ * model update.
+ *
+ * Assuming updating to the implied time is valid for the model, the function executes one or more model updates to
+ * process future time steps for the necessary indexes.  Multiple time steps updates occur when the given future time
+ * step index is not the next time step index to be processed.  Regardless, all processed time steps have the size
+ * supplied in `t_delta`.
+ *
+ * However, it is possible to provide `t_index` and `t_delta` values that would result in the aggregate updates taking
+ * the model's time beyond its `end_time` value.  In such cases, if the formulation config indicates this model is
+ * not allow to exceed its set `end_time`, the function does not update the model and throws an error.
+ *
+ * The function will return the value of the primary output variable (see `get_bmi_main_output_var()`) for the given
+ * time step after the model has been updated to that point. The type returned will always be a `double`, with other
+ * numeric types being cast if necessary.
+ *
+ * The BMI spec requires for variable values to be passed to/from models via as arrays.  This function essentially
+ * treats the variable array reference as if it were just a raw pointer and returns the `0`-th array value.
+ *
+ * @param t_index The index of the time step for which to run model calculations.
+ * @param d_delta_s The duration, in seconds, of the time step for which to run model calculations.
+ * @return The total discharge of the model for the given time step.
+ */
+double Bmi_Fortran_Formulation::get_response(time_step_t t_index, time_step_t t_delta) {
+    if (get_bmi_model() == nullptr) {
+        throw std::runtime_error("Trying to process response of improperly created BMI C formulation.");
+    }
+    if (t_index < 0) {
```

Comment:
I have seen models where negative time steps where real I think it was in DFLOW. Do we want to account for this possibility?

Reply:
Not in the scope of this PR 😄 .  It might be worth considering in an issue to relax the requirement in the future.